p2p/src/protocol.rs

```diff
@@ -251,15 +251,17 @@ impl MessageHandler for Protocol {
 					sm_req.hash, sm_req.height
 				);
 
-				let txhashset = self.adapter.txhashset_read(sm_req.hash);
```

Comment:
We need to think this through a bit. Our peer sends us a height and hash (the `sm_req`) but we are now effectively ignoring this and just sending the most recent snapshot.

I wonder if we need to take the height send by our peer and quantize this to the nearest previous snapshot (i.e. the closest `720` snapshot to that height).

Alternatively maybe we should be ignoring the values passed in by our peer - all we care about is serving the most recent snapshot?

Reply:
I thought about it for a while, but I don't think we would want to provide them any other txhashset. I considered verifying that the requested header is within an expected range, but it seemed unlikely enough to not justify any additional complexity.